zenml/core/datasources/base_datasource.py

```diff
@@ -96,7 +96,7 @@ def from_config(cls, config: Dict):
             config: a DataStep config in dict-form (probably loaded from YAML).
         """
         if keys.DataSteps.DATA not in config[keys.PipelineKeys.STEPS]:
-            raise Exception("Cant have datasource without data step.")
+            return None  # can be empty
```

Comment:
Just a small question to clear it for my understanding. Under which circumstances can this be empty?

Reply:
so two cases for now:

* Do print(pipeline) -> datasource.to_config() is used
* DeployPipeline -> Does not have a datasource